File: src/python/Providers/ProviderStrategy.py
class ProviderStrategy:
    def __init__(self, provider, model, api_key, queryString):
        self.provider = provider
        self.model = model
        self.api_key = api_key
        self.queryString = queryString

    def generate(self):
        raise NotImplementedError("Subclasses must implement this method")

File: src/java/Providers/ProviderStrategy.java
package src.java.Providers;

public abstract class ProviderStrategy {
    private final String provider;
    private final String model;
    private final String api_key;
    private final String queryString;

    public ProviderStrategy(String provider, String model, String api_key, String queryString) {
        this.provider = provider;
        this.model = model;
        this.api_key = api_key;
        this.queryString = queryString;
    }

    protected String getModel(){
        return this.model;
    }

    protected String getQuery(){
        return this.queryString;
    }

    protected String getApiKey(){
        return this.api_key;
    }

    public abstract String generate();
}